isb_web/isb_solr_query.py

```diff
@@ -0,0 +1,222 @@
+import typing
+
+import requests
+import geojson
+
+import isb_web.config
+
+BASE_URL = isb_web.config.Settings().solr_url
+_RPT_FIELD = "producedBy_samplingSite_location_rpt"
+
+# Identify the bounding boxes for solr and leaflet for diagnostic purposes
+SOLR_BOUNDS = -1
+LEAFLET_BOUNDS = -2
+
+MIN_LAT = "min_lat"
+MAX_LAT = "max_lat"
+MIN_LON = "min_lon"
+MAX_LON = "max_lon"
+
+# 0.2 seems ok for the grid cells.
+_GEOJSON_ERR_PCT = 0.2
+# 0.1 for the leaflet heatmap tends to generate more cells for the heatmap “blob” generation
+_LEAFLET_ERR_PCT = 0.1
+
+
+def _get_heatmap(
+    q: typing.AnyStr, bb: typing.Dict, dist_err_pct: float, grid_level=None
+) -> typing.Dict:
```

Comment:
Consolidated the heatmap querying code into a common method since it seemed to be identical.